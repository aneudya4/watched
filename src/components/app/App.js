import React, { useReducer, useEffect, useState } from 'react';
import AppRouter from '../routers/AppRouter';
import {
  DispatchContext,
  WatchListContext,
  AuthFormsContext,
} from '../../appContext';
import firebaseApp from '../../firebase';
import fetchMediaReducer from '../reducers/fetchMediaReducer';
import authFormsReducer from '../reducers/authFormsReducers';
import watchListReducer from '../reducers/watchListReducer';
import './App.css';

function App() {
  const [user, setCurrentUser] = useState(null);
  const mediaInitialState = {
    movies: [],
    genres: [],
    similarMovies: [],
  };
  const watchListInitialState = [];
  const authInitialState = {
    showLogin: false,
    showRegister: false,
    isAuth: null,
    user,
  };

  const [media, dispatch] = useReducer(fetchMediaReducer, mediaInitialState);
  const [auth, authDispatch] = useReducer(authFormsReducer, authInitialState);

  const [watchList, watchListDispatch] = useReducer(
    watchListReducer,
    watchListInitialState,
  );

  useEffect(() => {
    firebaseApp.auth().onAuthStateChanged(setCurrentUser);
  }, [user]);

  useEffect(() => {
    if (user) {
      authDispatch({ type: 'LOG_IN_USER', payload: user });
    }
  }, [user]);

  return (
    <AuthFormsContext.Provider value={{ auth, authDispatch }}>
      <DispatchContext.Provider value={{ dispatch, watchListDispatch }}>
        <WatchListContext.Provider value={watchList}>
          <div className="App">
            <AppRouter />
          </div>
        </WatchListContext.Provider>
      </DispatchContext.Provider>
    </AuthFormsContext.Provider>
  );
}

export default App;
