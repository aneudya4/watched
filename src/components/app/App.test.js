/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('Test on the <App/> Component', () => {
  it('renders App without crashing', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });
});
