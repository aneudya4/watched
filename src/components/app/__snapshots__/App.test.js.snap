// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on the <App/> Component renders App without crashing 1`] = `
<ContextProvider
  value={
    Object {
      "auth": Object {
        "isAuth": null,
        "showLogin": false,
        "showRegister": false,
        "user": null,
      },
      "authDispatch": [Function],
    }
  }
>
  <ContextProvider
    value={
      Object {
        "genres": Array [],
        "movies": Array [],
        "similarMovies": Array [],
      }
    }
  >
    <ContextProvider
      value={
        Object {
          "dispatch": [Function],
          "watchListDispatch": [Function],
        }
      }
    >
      <ContextProvider
        value={Array []}
      >
        <div
          className="App"
        >
          <AppRouter />
        </div>
      </ContextProvider>
    </ContextProvider>
  </ContextProvider>
</ContextProvider>
`;
